bluetooth/src/commonMain/kotlin/device/DefaultDeviceConnectionManager.kt

```diff
@@ -193,7 +193,10 @@ abstract class BaseDeviceConnectionManager(
     }
 
     fun handleUpdatedCharacteristic(uuid: UUID, succeeded: Boolean, onUpdate: ((Characteristic) -> Unit)? = null) {
-        notifyingCharacteristics[uuid.uuidString]?.updateValue()
+        notifyingCharacteristics[uuid.uuidString]?.let {
+            onUpdate?.invoke(it)
```

Comment:
Are you sure that onUpdate should be called before updateValue ?

Reply:
Yes